athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -186,10 +186,19 @@ public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesReque
                 .filter(index -> !index.startsWith("."));
 
         // get all data streams from ES, 1 data stream can contains multiple indices which start with ".ds-xxxxxxxxxx"
-        Stream<String> dataStreams = client.indices().getDataStream(new GetDataStreamRequest("*"), RequestOptions.DEFAULT).getDataStreams()
-                .stream()
-                .map(dataStream -> dataStream.getName());
-
+        Stream<String> dataStreams;
+        try {
+            dataStreams = client.indices().getDataStream(new GetDataStreamRequest("*"), RequestOptions.DEFAULT).getDataStreams()
+                    .stream()
+                    .map(dataStream -> dataStream.getName());
+        }
+        // gracefully exit for AWS ES.
+        // AWS's version of ES doesn't support data stream, it is a feature only available in the Elasticsearch distributed by Elastic itself, licensed under the Elastic License.
+        // https://stackoverflow.com/questions/68006918/aws-elasticsearch-service-and-datastream
+        catch (Exception ex) {
+            logger.warn(String.format("Unable to retrieve datastream or cluster version not support data stream, ignore datastream, message: %s", ex.getMessage()), ex);
+            dataStreams = Stream.empty();
```

Comment:
wondering if we should just init dataStreams to `Stream.empty()` on line 189 so you don't have to do this in the catch block. It would be best if we could just do this all in one line but I don't think there's a good way to do that. Only alternative would be to put all this data stream logic in a function and then just grab it on line 203 with getDataStreams() and let it handle returning either an empty stream or non-empty stream. In the interest of minimizing "how much" our functions do, I'm leaning towards that.